products/BellHybrid/services/evtmgr/screen-light-control/ScreenLightControl.cpp

```diff
@@ -143,55 +138,34 @@ namespace bell::screen_light_control
         ::screen_light_control::functions::setRampTarget(function.target);
     }
 
-    void ScreenLightController::setParameters(ManualModeParameters params)
+    void ScreenLightController::setParameters(const ConstLinearProgressModeParameters &params)
     {
-        brightnessValue = params.manualModeBrightness;
-        ::screen_light_control::functions::setRampTarget(brightnessValue);
-        setManualBrightnessLevel(brightnessValue);
-    }
-
-    void ScreenLightController::enableAutomaticMode()
-    {
-        if (lightOn) {
-            enableTimers();
-        }
-        automaticMode = ScreenLightMode::Automatic;
-    }
-
-    void ScreenLightController::disableAutomaticMode()
-    {
-        disableTimers();
-        automaticMode = ScreenLightMode::Manual;
-        automaticModeFunctions.clear();
+        setManualBrightnessLevel(params.targetBrightness);
     }
 
     void ScreenLightController::turnOn(const std::optional<ManualModeParameters> &params)
     {
         if (params.has_value()) {
-            const auto brightness = params->manualModeBrightness;
-            ::screen_light_control::functions::setRampTarget(brightness);
-            setManualBrightnessLevel(brightness);
+            brightnessWhenOn = params->manualModeBrightness;
         }
 
+        ::screen_light_control::functions::setRampTarget(brightnessWhenOn);
         bsp::eink_frontlight::turnOn();
-        if (automaticMode == ScreenLightMode::Automatic) {
-            enableTimers();
-        }
+        enableTimers();
         lightOn = true;
     }
 
     void ScreenLightController::setManualBrightnessLevel(bsp::eink_frontlight::BrightnessPercentage brightness)
```

Comment:
Can we change the naming? It sets the ramp params. ManualBrightnessLevel may be misleading.

Reply:
done